CRM/Utils/SQL.php

```diff
@@ -122,4 +122,27 @@ public static function isGroupByModeInDefault() {
     return TRUE;
   }
 
+  /**
+   * Does the database support multiple Locks.
+   *
+   * https://dev.mysql.com/doc/refman/5.7/en/miscellaneous-functions.html#function_get-lock
+   */
+  public static function supportsMultipleLocks() {
+    $version = self::getDatabaseVersion();
+    if (stripos($version, 'mariadb') !== FALSE) {
+      return FALSE;
+    }
+
+    return version_compare($version, '5.7.5', '>=');
+  }
+
+  /**
+   * Get the version string for the database.
+   *
+   * @return string
+   */
+  public static function getDatabaseVersion() {
+    return CRM_Core_DAO::singleValueQuery('SELECT VERSION()');
```

Comment:
First reaction -- this seems like something to cache in `Civi::$STATICS[__CLASS__]['version']`. But maybe it makes more sense to grep and see how often `acquire()` is used:

* Most calls to `acquire()` deal with backend jobs. The impact of a few ` select version()` queries seems pretty small in that context. This observation pushed me toward "Don't care/doesn't matter."
* There is an `acquire()` call in`CRM_Core_BAO_Cache::setItem()`. That seems like it might get hit multiple times, although I can't really say it's a big deal.

Reply:
Yeah - I'm leaning towards "Don't care/doesn't matter." since I don't have real skin in this - I implemented something a bit similar to this in Omnipay & put up a PR for core at the same time since this has been a missing fix for a while

Reply:
(I mean caching would make sense but I don't think this change needs to require that - it could happen at any point when someone feels like adding it)